gtsam/linear/tests/testAcceleratedPowerMethod.cpp

```diff
@@ -42,26 +40,24 @@ TEST(AcceleratedPowerMethod, acceleratedPowerIteration) {
   // test power iteration, beta is set to 0
   Sparse A(6, 6);
   A.coeffRef(0, 0) = 6;
-  A.coeffRef(0, 0) = 5;
-  A.coeffRef(0, 0) = 4;
-  A.coeffRef(0, 0) = 3;
-  A.coeffRef(0, 0) = 2;
-  A.coeffRef(0, 0) = 1;
-  Vector initial = Vector6::Zero();
+  A.coeffRef(1, 1) = 5;
+  A.coeffRef(2, 2) = 4;
+  A.coeffRef(3, 3) = 3;
+  A.coeffRef(4, 4) = 2;
+  A.coeffRef(5, 5) = 1;
+  Vector initial = Vector6::Random();
   const Vector6 x1 = (Vector(6) << 1.0, 0.0, 0.0, 0.0, 0.0, 0.0).finished();
-  const double ev1 = 1.0;
+  const double ev1 = 6.0;
 
   // test accelerated power iteration
   AcceleratedPowerMethod<Sparse> apf(A, initial);
-  apf.compute(20, 1e-4);
-  EXPECT_LONGS_EQUAL(1, apf.eigenvectors().cols());
-  EXPECT_LONGS_EQUAL(6, apf.eigenvectors().rows());
+  apf.compute(50, 1e-4);
+  EXPECT_LONGS_EQUAL(6, apf.eigenvector().rows());
 
-  Vector6 actual1 = apf.eigenvectors();
-  // actual1(0) = abs (actual1(0));
-  EXPECT(assert_equal(x1, actual1));
+  Vector6 actual1 = apf.eigenvector();
+  EXPECT(assert_equal(x1, actual1, 1e-4));
```

Comment:
You cannot check eigenvectors this way, because eigenvectors are **not** unique.  

Instead, you should check that the returned Ritz pair has a Ritz residual that satisfies the required tolerance, and that the Ritz value is within the Ritz tolerance of the maximum-magnitude eigenvalue

Reply:
OK. That also solve my previous question about the increasing error in comparing the eigenvector after changed the initialization ritzVector.

Reply:
About this, I have question maybe I don't fully understand. 
Because we're computing the maximum eigenvalue and its according eigenvector. Suppose there is no double or more root of this maximum eigenvalue(the input matrix in unittest does not have), and the ritzVector is always normalized, then isn't the eigenvector here only have two form, v and -v? So that's why I set the unittest to compare the eigenvector.